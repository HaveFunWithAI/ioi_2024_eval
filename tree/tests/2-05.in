2000
0 1 2 3 4 4 6 7 7 9 7 11 12 12 13 15 16 17 17 18 20 21 21 23 8 25 26 26 28 29 29 31 31 33 34 35 36 34 38 31 34 36 42 43 44 45 46 46 34 49 50 51 52 53 54 54 54 55 58 59 56 51 60 56 57 65 66 65 49 69 70 71 66 73 74 75 76 75 78 79 78 81 75 83 83 85 84 67 88 88 90 89 91 93 94 94 94 94 98 99 99 100 102 92 104 103 103 107 108 109 94 102 112 113 114 115 116 116 115 119 119 121 121 123 123 125 123 124 120 118 130 121 112 113 134 134 136 136 136 112 136 140 134 143 143 145 129 134 127 114 150 150 151 153 153 155 155 157 155 118 160 160 161 163 161 165 164 162 162 162 170 99 172 173 173 173 176 177 178 177 178 179 179 179 184 184 184 187 186 182 178 177 188 193 173 195 184 197 198 197 199 150 202 203 204 205 67 43 208 130 210 206 212 213 214 214 214 217 215 215 134 132 222 223 222 222 224 82 198 229 121 231 232 232 233 235 236 235 235 231 240 241 233 243 239 245 238 198 166 190 221 251 251 253 253 254 254 257 255 259 260 251 260 260 264 254 266 260 253 267 270 256 272 259 274 275 276 273 278 279 280 280 282 280 284 285 285 287 288 283 281 291 284 293 293 261 278 263 298 293 266 297 302 284 271 260 306 193 150 291 116 112 174 178 314 315 316 316 314 319 142 161 213 323 141 325 326 160 328 328 330 78 332 333 332 154 132 337 338 339 339 341 340 340 344 344 342 338 342 341 347 351 347 341 354 355 356 356 354 359 359 361 359 362 364 339 366 346 368 369 353 351 372 275 374 183 376 151 378 36 209 381 314 383 383 208 386 387 388 389 390 391 392 392 388 395 396 396 396 396 396 401 396 403 404 405 406 407 407 409 410 409 412 413 414 414 409 409 418 419 410 410 422 419 424 425 425 425 428 429 430 429 432 411 430 435 435 427 438 439 425 395 442 443 444 445 443 444 448 449 449 451 451 453 454 450 448 455 458 459 451 429 388 422 464 465 406 467 447 467 415 471 472 434 462 465 476 463 478 430 462 414 482 482 484 484 484 485 403 489 451 419 405 493 493 495 496 496 495 499 500 501 500 497 499 498 506 503 508 509 510 511 512 509 514 508 512 517 517 519 519 521 396 495 524 439 526 503 528 529 530 531 532 533 533 535 536 531 529 539 412 422 425 543 544 545 545 547 548 548 549 551 388 553 553 555 485 557 399 395 560 560 561 563 562 563 560 567 567 569 570 570 572 573 574 575 572 577 578 576 572 574 582 583 583 585 586 587 587 588 589 583 579 593 587 595 583 591 582 578 600 578 570 603 604 605 606 607 608 609 608 610 611 613 605 615 611 617 618 609 620 605 622 608 624 613 617 627 627 629 617 572 632 580 621 614 496 415 638 639 613 641 591 643 643 645 645 646 566 649 650 613 452 653 653 653 656 657 657 659 654 659 659 663 664 665 666 656 653 669 665 671 672 657 674 553 456 474 678 678 680 680 681 683 681 684 386 687 688 689 690 689 691 691 694 695 696 695 695 699 692 690 690 701 692 703 706 707 708 709 706 706 709 713 714 714 715 717 718 718 713 721 719 723 698 694 726 727 727 729 699 709 732 733 734 734 733 737 736 737 740 740 742 742 744 744 744 747 748 742 750 751 752 752 752 753 756 754 750 759 745 761 742 751 759 755 766 767 768 769 770 745 772 773 773 775 774 775 776 776 772 781 782 783 781 785 785 787 786 782 774 777 783 793 794 791 796 772 798 799 777 801 802 801 804 804 806 807 808 808 810 807 799 754 764 762 816 816 816 819 692 821 822 787 824 824 769 757 561 829 737 831 832 833 833 586 504 837 837 839 839 838 839 843 844 845 844 843 848 849 850 851 852 853 853 853 856 856 857 853 859 861 862 853 864 854 866 865 858 858 859 849 868 873 854 853 876 877 864 855 875 881 882 883 879 885 886 887 887 889 845 891 892 892 893 895 894 897 898 899 897 899 837 903 904 904 904 907 908 908 909 906 912 913 914 912 916 917 917 919 920 919 922 923 923 922 903 915 928 929 929 903 932 933 934 935 936 937 938 938 940 937 935 942 937 937 946 939 905 949 950 908 952 953 912 955 956 919 958 941 960 961 960 963 903 903 966 939 968 968 970 971 905 926 974 915 941 977 910 979 980 968 931 979 966 985 986 986 987 989 990 990 991 990 917 995 995 997 997 995 1000 1000 999 922 1004 984 1006 1006 1008 972 684 1011 546 1013 1014 1013 1016 1017 1018 1019 1020 1021 1021 1017 1024 1025 1020 1027 897 951 1030 1031 1032 1033 1034 1035 1035 1036 1035 1039 1040 1041 1042 1043 1044 1034 1036 1047 1038 1049 1049 1049 1052 1051 1054 1052 1050 1057 1042 1059 1051 1032 1062 1063 1063 1065 1066 1062 1042 1069 1070 1050 1072 1073 1031 1075 1076 1077 1037 1079 1079 1080 1080 1082 1039 1085 1086 1086 1088 1086 1083 1091 677 1093 1093 1095 1095 513 1098 689 869 1101 748 742 1104 1105 770 1032 1108 1108 1110 1111 1112 1109 503 1115 1115 1117 1118 1116 1120 1120 1122 1123 1124 1124 1124 1122 1128 1129 1128 1120 1132 1133 1130 1135 1135 1137 1128 1118 1140 1121 1142 1142 1144 1145 1145 1147 1146 1149 1150 1129 1152 1152 1152 1154 1156 1127 1158 1158 1158 1161 1161 1162 1160 1160 1164 1167 1168 1169 1169 1171 1163 1166 1174 1175 1175 1177 1178 1116 1162 1181 1182 1182 1182 1185 1185 1187 1188 1187 1188 1191 1191 1193 1192 1193 1193 1181 1198 1198 1187 1201 1202 1202 1202 1205 1204 1198 1193 1191 1210 1211 1198 1213 1213 1194 1185 1214 1203 1219 1194 1190 1222 1223 1218 1198 1226 1227 1226 1226 1230 1231 1232 1233 1232 1226 1236 1237 1238 1238 1240 1240 1242 1243 1244 1245 1244 1247 1190 1249 1250 1250 1177 1181 1248 1121 1256 1256 1258 1259 1260 1256 1262 1262 1264 1256 1266 1258 1268 1269 1270 1271 1272 1265 1274 1275 1274 1273 1278 1279 1269 1260 1265 1262 1284 1285 1284 1287 1288 1289 1290 1289 1292 1292 1293 1293 1292 1297 1289 1287 1300 1301 1300 1303 1304 1305 1306 1305 1305 1309 1309 1307 1308 1313 1311 1302 1306 1317 1303 1259 1320 1321 1322 1323 1323 1325 1326 1325 1325 1325 1325 1331 1332 1321 1326 1335 1336 1332 1334 1339 1340 1325 1342 1286 1344 1345 1310 1347 1348 1349 1350 1351 1351 1352 1347 1355 1313 1357 1270 1359 1359 1361 1362 1361 1364 1364 1360 1361 1368 1369 1361 1259 1372 1373 1321 1345 1331 1377 1377 1379 1378 1381 1382 1362 1381 1182 1386 1387 1387 1388 1150 1391 1384 1275 1169 1297 1383 1293 1117 1399 1400 1401 1402 1150 1404 1405 1405 1406 1211 1409 1327 1282 1412 1412 1414 1414 1413 1417 1417 1419 1420 1418 1414 1423 658 1425 1426 1426 1425 1426 1430 1429 1432 1433 1433 1430 1436 1436 1438 1439 1440 1439 1442 1443 1441 1445 1446 1446 1448 1446 1450 1447 1452 1447 1454 1455 1454 1444 1443 1441 1460 1460 1462 1463 1463 1465 1465 1466 1462 1469 1470 1462 1458 1436 1447 1475 1476 1475 1475 1479 1476 1481 1482 1479 1484 1484 1486 1487 1486 1489 1485 1491 1492 1493 1493 1452 1496 1496 1496 1461 1500 1501 1455 1479 1475 1457 1469 1460 1508 1481 1464 1511 1511 1512 1514 1515 1514 1512 1513 1519 1520 1521 1522 1522 1524 1525 1526 1525 1519 1529 1446 1531 1531 1531 1533 1535 1535 1537 1538 1533 1539 1534 1536 1543 1537 1471 1534 1547 1547 1549 1550 1550 1552 1553 1554 1553 1552 1533 1558 1559 1559 1558 1562 1563 1563 1565 1566 1565 1568 1569 1569 1560 1564 1573 1574 1574 1561 1567 1578 1557 1463 1581 1582 1582 1582 1582 1586 1586 1583 1587 1590 1586 1592 1593 1594 1592 1596 1594 1598 1598 1599 1601 1602 1603 1603 1604 1594 1594 1608 1587 1610 1611 1592 1610 1614 1615 1615 1587 1598 1560 1437 1621 1622 1621 1624 1625 1622 1627 1627 1629 1630 1631 1632 1633 1634 1632 1636 1637 1633 1639 1640 1640 1641 1634 1644 1645 1644 1645 1648 1635 1650 1651 1651 1653 1654 1653 1655 1656 1651 1659 1630 1655 1635 1663 1663 1664 1641 1667 1668 1667 1667 1461 1672 1673 1674 1675 1676 1676 1677 1676 1674 1673 1682 1683 1672 1680 1672 1687 1688 1688 1687 1684 1683 1693 1694 1695 1695 1694 1698 1691 1700 1701 1646 1703 804 1705 1706 1032 1708 434 1710 1141 1712 1713 1712 1715 1573 1717 1718 1150 1720 1720 1031 1671 1563 1725 1688 1727 1728 1728 1728 1728 1732 1134 1734 1734 1736 1737 1738 1739 1735 725 1742 1743 1744 1745 1746 1747 1748 1742 1750 1745 1752 1747 1745 1755 1755 1755 1756 1759 1759 1758 1762 1756 1757 1765 1765 1767 1765 1766 1770 1771 1771 1773 1765 1774 1776 1776 1777 1779 1756 1775 1782 1783 1784 1785 1784 1786 1788 1789 1790 1782 1792 1793 1794 1795 1792 1794 1784 1799 1800 1799 1802 1803 1800 1805 1801 1807 1791 1809 1782 1811 1812 1813 1812 1815 1815 1817 1818 1813 1820 1820 1822 1819 1824 1825 1826 1823 1828 1785 1812 1831 1832 1832 1813 1832 1836 1836 1838 1839 1840 1840 1840 1842 1842 1836 1837 1847 1821 1805 1850 1824 1852 1769 1854 1855 1855 1854 1855 1859 1827 1774 1862 1863 1864 1865 1862 1867 1867 1869 1869 1862 1871 1873 1874 1873 1875 1877 1874 1873 1877 1881 1882 1873 1875 1875 1870 1887 1888 1867 1890 1891 1891 1862 1894 1891 1830 1897 1850 1899 1830 1901 1901 1903 1903 1904 1902 1902 1908 1909 1906 1911 1905 1913 1914 1915 1913 1917 1786 1782 1920 1921 1882 1923 1924 1923 1926 1857 1928 1928 1928 1928 1931 1928 1932 1935 1936 1937 1938 1937 1940 1940 1941 1934 1944 1945 1946 1945 1933 1949 1950 1931 1952 1818 1792 1955 1956 1957 1839 1785 1870 1754 1181 1287 1964 1965 1424 1967 1967 1968 1967 1967 1781 1973 1973 1975 1289 1977 1444 1828 620 1981 1982 1981 1984 1985 1986 1985 1983 1061 1143 1991 1991 1993 1991 1993 1996 1996 1998
901643 293058 273565 823398 862619 922649 320055 690451 533300 152438 99697 725025 74734 603349 877712 789881 814473 946916 654774 895098 498328 927753 966551 900643 295777 588207 238020 916168 251028 366670 125881 921907 86757 717072 222143 622043 133678 751385 454100 621575 436812 428749 159423 646474 909680 465599 893996 935292 717066 521707 898762 187658 1329 878097 703223 892095 766099 547323 512538 916284 665104 284607 578888 818953 264437 555188 796144 878089 968879 206383 992365 117093 893175 115689 455358 828686 870292 968921 638604 905683 228701 276367 303532 978087 901914 171056 747438 685 704955 389433 909067 213986 950956 704919 704929 55531 42004 44305 290032 421842 511190 580792 634528 654893 359508 959713 115843 967926 265795 475831 106244 42505 820073 219861 454059 510249 258498 537519 11555 213795 373665 328051 825321 463734 994896 554701 921920 537642 695001 717918 920062 960630 687003 490992 201729 199040 842627 47411 362436 57407 49356 560130 99480 762651 15645 436041 703256 705672 416390 231221 360415 31429 271010 88978 281887 609272 473357 171158 993764 159390 484782 943079 981972 25761 970300 949126 862516 599184 723578 792079 843514 943423 851283 146775 349907 127377 639290 372742 368238 935962 360732 765702 117233 132937 396109 36513 373267 659387 53771 931681 242169 201589 36665 920128 827579 441456 587966 328454 665090 220058 547543 694678 156971 629566 529974 756843 751344 772999 435387 775129 507980 182965 566105 940729 261960 344348 220173 861341 781846 270254 419188 436513 359100 22423 843510 248382 772228 879244 456129 829116 973072 40815 817053 785819 664054 233774 592456 96672 341858 361838 53712 884666 979995 342615 349691 389914 161934 130091 761501 909626 29737 314260 924219 93975 989189 585195 150163 713178 609012 145469 99140 758136 396192 178188 207594 367085 358293 782746 78937 597519 872077 965077 603167 350889 542573 801264 100229 374412 626286 484518 381870 533297 81337 169884 581151 276759 487372 283173 766650 395709 225941 398734 697545 651172 814141 140296 466673 588621 562042 280087 429926 237607 395818 368696 198319 520807 479785 270339 753108 979524 137605 426607 246147 379160 736379 893122 40902 137129 15406 264960 753080 528803 681848 936444 78191 453700 587926 523041 621163 472675 76832 789047 617077 463508 692843 499040 802600 264200 578289 948511 560515 326133 885922 486789 830980 70 81514 48199 65650 151336 381663 917118 770010 230076 306952 944095 94522 291471 206622 79173 421940 277649 579027 543990 950385 624353 608412 662275 406470 470305 569255 828785 537103 157675 241628 538815 970400 950918 472809 480542 901982 49629 764467 762807 924417 667489 261811 263961 407207 477050 231584 33124 203795 359250 707744 453395 932550 959113 518934 72847 596841 477760 14975 16079 183229 267542 602796 263968 997055 894038 607979 811891 902764 167646 492009 553371 682870 594617 835623 118082 135431 981518 726323 974271 151476 193250 746657 720669 417846 289531 694946 153265 74701 353066 554380 561312 132241 263031 408737 341301 579915 922 619619 308300 474133 38664 93483 865283 333421 723418 303482 1244 967393 181966 846802 860209 553186 151701 789244 824854 527168 874112 435628 135116 341574 766435 288020 750137 141635 179955 700395 800037 562010 682245 88669 143984 971220 403311 241957 367968 964687 798916 64840 926846 807298 739223 688195 867327 644515 901108 42314 353042 308523 933694 5171 971770 801211 301616 596488 366267 36564 114998 709927 490437 444908 608122 425098 240521 831261 326445 85435 946056 726487 89053 314042 129559 291831 997036 353613 414008 350486 917708 873095 976392 706515 746941 565999 437302 252190 14155 654522 105694 654269 583547 27820 425876 908189 580275 164771 522553 185021 799404 75280 135005 986417 585055 938483 768502 66602 502199 219086 756575 273608 111194 940392 119891 466812 586185 557161 10579 419424 699993 787847 799036 587187 294486 134508 192332 622455 372702 475797 656568 583456 184123 225594 186134 184314 550311 43636 218200 509390 18861 372582 814923 207630 169261 713327 508304 66735 835589 546561 148948 522383 281000 34805 511952 553807 465783 254254 660717 666856 632103 558468 703649 443322 297833 998915 641828 823062 218653 227017 337195 238451 930317 354657 989456 440136 326312 934541 510516 919616 450710 257221 913304 926752 883092 542400 220634 6994 584518 73269 313066 249351 409813 386962 990232 620885 438739 17758 969175 462180 893021 214064 71929 939137 222904 755033 920118 560270 463015 31069 551438 614812 570104 277559 566451 519372 169968 684714 72588 818195 317029 622116 868767 549943 883094 67240 575889 392416 89363 738412 560119 327676 101369 920786 698620 684746 767949 664212 145574 699503 88746 833515 720752 84492 127745 736064 908317 741385 989634 646202 116984 445548 134826 461177 367542 85245 224239 27497 679816 425701 411317 124367 752486 65767 756251 640488 288145 334628 986973 692496 30797 739801 730386 880415 103907 845465 496360 897823 300360 429061 260148 26056 482708 839712 305503 691655 245108 643800 613036 329357 837687 330528 74703 686579 208781 804417 70839 544683 448702 390614 559185 207819 73412 241444 658121 944926 419763 852210 942009 482403 151283 295032 270678 218643 330853 541033 437565 119962 348352 213624 167180 39232 942088 431619 142247 136708 993367 795542 732719 548701 46376 863315 161326 61477 377732 715449 765650 575715 66092 599305 384303 913731 680876 582697 477780 803476 681764 834357 355313 196985 943593 125025 863330 303027 771159 331152 530877 59039 266168 2193 225973 134794 770086 133801 404387 188802 789654 985668 832529 261688 518500 825675 554542 131820 159102 817402 581713 480729 333312 251479 526584 961523 476277 691961 590010 465584 704917 791405 385680 625727 446458 979149 282298 244700 909264 388738 247053 137902 219271 586241 298226 51117 172249 154322 771974 722831 558435 197242 567720 130883 407120 377993 417775 368533 910346 922743 483170 174538 820928 473708 205507 243210 575146 786488 55387 220156 466340 283665 963184 575394 718797 434951 998017 570507 938820 942905 191890 431461 30511 348904 644518 100971 907770 10383 521699 328529 905838 187185 592636 667220 313764 595557 235099 291526 859044 136367 68285 628672 104589 735301 378696 925957 185506 577900 471473 622715 623639 246451 756259 389487 18554 211630 951326 33173 989110 910120 968343 59814 423205 574907 118795 102263 801694 125267 711653 234374 421608 107775 944513 606485 191144 102346 702506 23610 344710 90423 58940 584204 144187 240205 83596 86102 652767 548435 59314 371512 88787 300323 372400 220467 269721 931511 840762 271672 579710 895100 414487 373357 201381 732518 167982 300857 629207 296795 771489 280658 637409 655194 133277 608820 623682 200510 164876 360559 495111 164920 189134 798629 617930 903394 557119 397809 642539 9966 764648 243803 583515 393952 196398 575910 485559 341668 182764 782868 843998 25111 975817 794700 206634 863428 885472 586336 543348 217293 247130 359331 793450 913484 284126 779877 529554 296252 259215 84437 149864 271046 926519 947624 450849 155402 428712 55116 148315 713213 522957 899296 307269 196954 200507 343374 53600 470498 573434 975365 459266 276018 240783 732708 941846 203056 958879 946885 587297 532521 248663 87572 849773 602670 176935 496682 102443 842167 728196 685674 613748 464175 440788 402906 585780 652202 880533 847791 41223 555327 640708 882255 332818 706012 696520 46796 523594 999618 793516 790359 997802 781947 306252 959134 765582 571123 907131 779333 20690 411668 88744 911783 340548 395045 256223 470286 983087 224177 611443 17121 863666 157434 314594 608439 587494 417229 851332 977038 25576 554698 532726 538709 180265 58131 731355 543699 889598 871932 288695 897771 965996 234844 60993 607394 962106 826619 348949 988244 562980 199145 823073 831399 131278 882908 614869 655262 521322 792250 75463 501437 875785 617092 771254 620557 686062 307794 28486 753186 465863 164889 586722 531785 963659 400371 81357 644924 95416 636095 757299 618928 792191 982285 474528 371439 719619 597566 385948 327018 489933 693489 450263 515729 370985 676862 179833 197625 577970 316378 699113 781598 558543 651939 552989 105315 752362 251672 936374 359516 206416 354035 758781 752272 514181 342584 434395 574522 998012 289802 450986 271801 99331 878629 343526 369077 101257 505102 702311 320717 504744 55431 234465 697777 89262 151451 320352 857260 609532 246257 532328 22319 711101 762495 964851 483410 60746 411903 933349 240338 517945 52760 439378 418693 108725 463026 874443 807736 651125 475192 859811 923441 650503 689274 14345 97338 679037 203022 409123 303858 808546 530445 457641 435182 158301 800140 27586 325611 256627 969900 438519 141592 714377 979202 136890 166580 27104 602504 90601 230201 918551 554530 244214 503579 483780 786902 705084 38142 43659 543106 998758 769959 352629 80948 497314 266425 630986 970041 34683 180429 59352 103518 491761 514620 869440 230682 517003 875365 68792 51134 171678 556836 282249 378039 528514 156047 858160 492921 495925 540430 659196 412136 858513 647149 166470 394464 711244 714213 237235 747063 242979 735105 500656 809160 454486 831148 3220 950577 354743 376253 922730 813017 739074 746414 296367 500302 759703 918905 859220 782475 808818 658717 531467 918954 917307 487366 579353 85253 852277 169969 871030 420874 228009 811420 850091 587736 816273 709513 780783 261158 324860 158586 336052 894240 141046 664801 419215 156185 326164 217438 178758 134815 141463 378113 849295 647977 183427 635817 30952 182080 148862 187424 561235 976055 636565 769833 2197 707159 154486 60440 809989 809734 264957 781661 802195 555757 423260 393437 134504 528702 437299 996001 965528 471696 495282 568201 478791 504202 299066 973990 509486 672962 750257 590718 260945 930212 435025 294872 764089 519886 428810 902706 574003 44948 962863 589547 791065 761707 784702 796460 404992 959848 248553 372519 285156 549050 777502 968179 354139 161251 105110 589605 784887 383641 45208 5830 943599 198401 233329 195897 812960 566412 549233 267604 9830 108580 190103 615669 353469 763814 832685 648591 369286 134040 502835 292722 896575 280149 854462 611025 622632 262091 922795 703728 512998 753216 416847 500994 701252 632893 372819 181131 295330 769695 365580 535745 619358 491029 243794 65697 783223 561160 97582 307493 411774 710285 861456 71622 355559 180412 39842 34041 224770 389233 17699 102772 117214 754022 275326 556557 210269 478075 494974 900111 662205 361926 301236 309572 240160 735127 504419 927064 841480 922368 384289 276596 154357 168030 600410 504833 101191 751527 899560 980601 943251 894558 341972 760802 313230 437746 503892 308580 581881 455611 261443 261415 189328 977887 63803 861875 564886 830302 110052 75623 71686 376247 913606 57897 678476 517203 831283 855596 530238 935656 186323 930104 433665 218961 240959 6520 284260 573674 208661 654451 697134 365686 534571 707896 17995 802639 161850 706988 824021 909650 496471 745855 739067 592458 108171 263934 79616 642298 805624 572490 608719 775410 583044 226304 283649 815358 282993 143547 987873 586454 54831 156650 50012 418394 110114 745885 925535 862083 216557 122516 242340 587287 854984 297811 342792 568139 385107 868422 465581 819489 417551 353824 587182 928399 520429 857687 460639 582097 896981 571689 604910 804421 140203 485109 262353 237409 507281 799598 645843 562150 191491 933433 800653 826811 613635 7218 580952 98607 125107 883879 215703 354075 536019 174521 427920 291804 827404 462986 447379 13404 789532 870935 54662 491823 79155 508033 339705 685509 99892 735201 275950 686711 818981 235258 916210 285694 965365 118474 970815 249737 659541 953902 586199 329403 892043 985683 100141 773401 537203 625364 808140 8823 86038 464919 95115 590394 567632 274729 868638 740802 201735 720101 716813 582734 650094 176237 18759 161285 94427 51916 501310 654641 890623 712031 264537 777493 350664 802366 248014 831749 358034 324850 874631 708759 400147 908799 189708 658089 54234 153304 323030 438305 840090 235853 13173 788297 174517 828513 790196 422110 396004 112097 197978 12827 999498 745231 197070 874249 493223 321347 153770 772366 177047 133728 399591 301168 843260 811917 878883 221981 606877 83935 904353 331278 267752 807458 152147 624275 484258 387642 96323 394133 549637 850549 912833 625527 560410 942729 995547 493823 826051 484733 231970 833562 7638 765689 249024 827344 927827 192958 858720 976730 611924 98620 364559 661357 688886 917069 109520 289104 367540 835803 107187 658377 366603 573891 338034 109160 719710 3378 987891 739048 204101 710919 5744 562737 555895 37163 334950 600932 600606 649913 645746 269260 499786 698889 931029 908957 753155 446872 438844 894118 769355 960404 437091 69800 263760 947896 560217 979082 605257 695441 69861 189681 584062 142487 812542 676345 616326 112571 795301 943644 788956 686068 853915 533653 728632 86502 69847 800787 844670 283470 837606 185122 268109 422009 665819 123736 443702 117798 729001 416581 905016 249079 196791 483582 862414 491214 21987 257423 174075 815258 1148 174432 513961 431020 976357 494765 262391 460529 63156 231107 264921 596895 312123 868466 826440 689843 792545 862793 66837 197609 466017 169332 412349 919097 51287 829847 100846 237190 748840 77196 560962 907206 951891 26738 643009 905336 970456 240922 790926 122906 275207 974847 428398 811727 103181 113419 364939 427941 254302 789050 571164 255693 656628 794199 431217 578841 304871 458272 367229 687613 93046 195804 530605 729950 388025 201499 580581 446034 118268 821120 333532 538219 83800 868371 208755 864347 340871 321532 957186 149840 982552 5323 145852 280298 641356 932107 755470 917658 372270 677416 131468 418442 757106 38022 563031 863404 378727 274400 184610 447672 600615 761002 86376 801338 120120 170588 335618 682439 698582 116421 341465 556248 110445 915344 632640 402329 125438 600755 392801 901897 876440
10
262521 325952
74462 96868
144462 194976
39294 355980
750068 959263
281533 367826
69352 211785
180022 788239
398189 473779
337182 392202
