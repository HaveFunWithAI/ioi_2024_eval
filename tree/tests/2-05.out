190697287624219
53603131790026
103217577508362
19182529285386
541772654508376
202493818900847
40634954006094
98609882258122
291520925855683
247847606357154
